streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -145,8 +146,12 @@ public StreamTask(final TaskId id,
 
         stateMgr.registerGlobalStateStores(topology.globalStateStores());
         if (eosEnabled) {
-            this.producer.initTransactions();
-            this.producer.beginTransaction();
+            try {
+                this.producer.initTransactions();
+                this.producer.beginTransaction();
+            } catch (final ProducerFencedException fatal) {
+                throw new TaskMigratedException(id, fatal);
```

Comment:
Update javadoc accordingly?

Reply:
Ditto below.

Reply:
Done. -> that's why I want to use checked exceptions... just saying \cc @dguy @bbejeck